File: frontend/src/services/legal.ts
import { api } from "@/api/client";
import type { LegalMetadata } from "@/types";

export const legalKeys = {
  metadata: ["legal", "metadata"] as const,
};

export const legalService = {
  /** Current versions + effective dates of the legal documents (public). */
  metadata: () => api.get<LegalMetadata>("/legal"),

  /** Record (re-)acceptance of the current Terms and/or Privacy Policy. */
  accept: (body: { terms_and_conditions: boolean; privacy_policy: boolean }) =>
    api.post("/auth/accept-legal", body),
};

File: backend/routers/legal.py
"""Public legal-document metadata (``/api/legal``).

Serves only the current version + effective date of the Terms of Service and
Privacy Policy — the document text itself is rendered by the frontend
(``frontend/src/content/legal/*``). Public and unauthenticated so the
registration and standalone /terms, /privacy pages can label which version a
visitor is about to accept before they have an account. Versions are the
single source of truth in ``backend/legal.py``.
"""

from fastapi import APIRouter

from ..legal import legal_metadata

router = APIRouter(prefix="/api/legal", tags=["legal"])


@router.get("")
def get_legal_metadata():
    return legal_metadata()
